java/ql/src/experimental/Security/CWE/CWE-094/SpringImplicitViewManipulation.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @name Spring Implicit View Manipulation
+ * @description Untrusted input in a Spring View Controller can lead to RCE.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id java/spring-view-manipulation-implicit
+ * @tags security
+ *       external/cwe/cwe-094
+ */
+
+import java
+import SpringViewManipulationLib
+
+private predicate canResultInImplicitViewConversion(Method m) {
+  m.getReturnType() instanceof VoidType
+  or
+  m.getReturnType() instanceof MapType
+  or
+  m.getReturnType().(RefType).hasQualifiedName("org.springframework.ui", "Model")
+}
+
+private predicate maybeATestMethod(Method m) {
```

Comment:
@aschackmull did I understand you'd prefer to switch to a uniform way of classifying probably-test files?

Reply:
When at some point we look into promoting this from experimental to standard, then yes, but for now this ad-hoc check is ok.